### PR TITLE
Updated axes_box_aspect.py and angle_annotation.py to regularize formatting

### PR DESCRIPTION
**Pull request by aseriesof-tubes**
## PR summary

Two files [axes_box_aspect.py](https://github.com/matplotlib/matplotlib/blob/78e366195deef21b0fac2c1b19c7b9b3dff9908a/galleries/examples/subplots_axes_and_figures/axes_box_aspect.py) and [angle_annotation.py](https://github.com/matplotlib/matplotlib/blob/78e366195deef21b0fac2c1b19c7b9b3dff9908a/galleries/examples/text_labels_and_annotations/angle_annotation.py) have been updated to meet formatting standards. 

This is for issue #26862 

## PR checklist

- [x] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [ ] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [ ] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [ ] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [x] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

**Review by QuLogic — DISMISSED** [review]

File: galleries/examples/subplots_axes_and_figures/axes_box_aspect.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 
 # %%
 # A square axes, independent of data
-# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (QuLogic) [inline galleries/examples/subplots_axes_and_figures/axes_box_aspect.py:18]: Looking at some other examples, I think these appear to be nearly top-level, and so really should be `=`.
```suggestion
# A square axes, independent of data
# ==================================
```
etc.

Comment (aseriesof-tubes) [reply galleries/examples/subplots_axes_and_figures/axes_box_aspect.py:18]: Understood! I'll be changing `^` to `=`, though it does seem like some other examples instead use `-`. Please confirm this.  Additionally, there seems to be an issue. In Axes box aspect, there seem to be two sections of the same name (Square joint/marginal plot). How should I resolve this?

Comment (aseriesof-tubes) [reply galleries/examples/subplots_axes_and_figures/axes_box_aspect.py:18]: Got it! I'll change the title to `Set data aspect with box aspect`. As for the formatting, they have been all corrected to `-` for the sake of consistency.

**Review by QuLogic — APPROVED** [review]

**Comment by QuLogic** [comment]
Thanks @aseriesof-tubes! Congratulations on your first PR to Matplotlib :tada: We hope to hear from you again.

**Comment by aseriesof-tubes** [comment]
Thank you for merging @QuLogic, I will definitely try my best to keep contributing to Matplotlib :)

**Pull request merged**